sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -844,20 +844,18 @@ private case class GpuOrcFileFilterHandler(
       OrcProto.Stream.Kind.ROW_INDEX)
 
     def getOrcPartitionReaderContext: OrcPartitionReaderContext = {
-      closeOnExcept(orcReader) { _ =>
```

Comment:
Why do we not want to close the reader if an exception occurs?  How is it guaranteed that this reader will eventually be closed?

Reply:
the caller who will close orcReader if exception happens

Reply:
And updated the description for the PR